services/indexes/pvm/writer.go

```diff
@@ -706,14 +708,23 @@ func (w *Writer) InsertTransactionBlock(ctx services.ConsumerCtx, txID ids.ID, b
 func (w *Writer) InsertMultisigAlias(
 	ctx services.ConsumerCtx,
 	alias *multisig.Alias,
-	changeAuth verify.Verifiable,
+	auth verify.Verifiable,
 	txID ids.ID,
 ) error {
 	var err error
 
-	// If changeAuth is nil, then delete all aliases for this multisig
-	if changeAuth == nil {
-		// Delete any existing multisig alias first
+	// If alias.ID is an empty ID, then it's a new alias, and we need to generate the aliasID from the txID
+	if alias.ID == ids.ShortEmpty {
+		alias.ID = multisig.ComputeAliasID(txID)
+	}
+
+	_, err = ctx.Persist().QueryMultisigAlias(ctx.Ctx(), ctx.DB(), alias.ID.String())
+	if err != nil && err != dbr.ErrNotFound {
+		return err
+	}
+
+	// if there is an already existing alias with this aliasID or auth is nil, then we need to delete it
+	if auth == nil || err != dbr.ErrNotFound {
```

Comment:
wouldn't it more comprehensible if the check was
` if auth == nil || **err == nil** { `